src/imgui/ImGuiReverseBar.cc

```diff
@@ -76,46 +98,84 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 						}
 					});
 				}
-				ranges::sort(stateNames, std::greater<>{}, &StateNames::ftime);
 			}
 			if (stateNames.empty()) {
 				ImGui::TextUnformatted("No save states found"sv);
 			} else {
 				im::Table("table", 2, ImGuiTableFlags_BordersInnerV, [&]{
```

Comment:
Another idea was to add a 3rd column with small previews of the screenshots. And only show the large (original sized) screenshot when hovering over the small one. This will also save a lot of space in the sub menu.

Reply:
Yeah, but I'm not sure I like it. IMHO it doesn't add much to the current solution. Can you really see something in such a small preview anyway?

Reply:
What it adds is that it saves _a lot_ of horizontal space. E.g. currently the menu is not usable in full-screen, scale_factor=2, it's wider than 640 pixels, it will never fit on screen. (We have other windows that are as wide or wider, but AFAIK only debug windows, none that you'd want to use in full-screen mode).

You don't loose much: currently you only see one screenshot at a time, and you must first click to see it. In this proposal you see a small preview of many savestates, and, if you want, also one big one (by hovering instead of clicking).

I didn't clearly say this, but the small preview I had in mind could be something line 32x20 pixels (that should already be reasonably recognizable) So taller than one normal text line. But that's a tradeoff, it means fewer rows are visible at the same time. (Very similar like in a file-browser with previews of your image (e.g. jpg) files).

Reply:
When going to full-screen, the native resolution of the screen is used. So the scale factor doesn't matter. The menu is usable fine in full-screen on my machine. But if you have a really low resolution screen it probably won't fit indeed.
We can also make the column with the name smaller and horizontally scrollable. Would that be helpful?

Also, currently you don't have to click to see the screenshot. It is already show in the preview window if you only hover on the name in the list/table. If you click, you already load the savestate.

Reply:
An alternative is to not have a 3rd preview column, but instead show the full-sized screenshot as a popup, when hovering over the name.
Though this has the disadvantage that the popup will block (most of) the rest of the table. Maybe this can be mitigated by using the `ImGuiHoveredFlags_NoSharedDelay` flag, see the documentation in `src/3rdparty/imgui/imgui.h`, lines 1379-1386.

Reply:
We discussed on IRC that we can leave it as it is now, until we find an application where the 640 pixel width is important.